framework/PostProcessorFunctions/HistorySetDelay.py

```diff
@@ -0,0 +1,91 @@
+
+"""
+This is to implement a delay or lagged parameters in a HistorySet
+"""
+
+import copy
+import numpy as np
+
+from PostProcessorInterfaceBaseClass import PostProcessorInterfaceBase, CheckInterfacePP
+from utils import InputData, InputTypes
+
+class HistorySetDelay(PostProcessorInterfaceBase):
+  """
+  Class to get lagged or delayed data out of a history set.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super().getInputSpecification()
+    inputSpecification.setCheckClass(CheckInterfacePP("HistorySetDelay"))
+    delayClass = InputData.parameterInputFactory("delay", InputTypes.StringType,
+                                                 descr="Adds a delay variable")
```

Comment:
verbosity of the description

Reply:
those descriptions will be used for future automatic documentation. they must be of "manual grade"